### PR TITLE
Rewrite requesttoken.spec.js with Jest

### PR DESCRIPTION
**Pull request by francoisfreitag**
[Jest](https://jestjs.io/) is a test runner that focuses on simplicity.
It instruments babel to transform modules and test them.

Using Jest simplifies the existing configuration and allows dropping a
bunch of workarounds, as well as following the shared Babel
configuration for new code.

Signed-off-by: François Freitag <mail@franek.fr>

Closes #27229 
Closes #27227

### CODE REVIEW

File: tests/jestSetup.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 /**
- * @copyright 2018 Christoph Wurst <christoph@winzerhof-wurst.at>
+ * @copyright 2021 François Freitag <mail@franek.fr>
  *
- * @author Christoph Wurst <christoph@winzerhof-wurst.at>
- * @author John Molakvoæ <skjnldsv@protonmail.com>
+ * @author François Freitag <mail@franek.fr>
```
Comment (skjnldsv) [inline tests/jestSetup.js:4]: Nope, the file is still owned by `Christoph Wurst`
You can add yourself tot he authors :)

**Review by skjnldsv — APPROVED** [review]
Yes please!! :heart:

**Comment by skjnldsv** [comment]
Can't thank you enough! It was really becoming complicated with webpack5 and stuff!! :rocket:

**Comment by francoisfreitag** [comment]
Thanks for the review!

I just downgraded the jest-babel and jest version to `26.6.3`. The main reason it to make it compatible with `vue-jest`. The 27.x series change the transformer interface in jest-babel. That causes vue tests from #27008 to fail with:

```
 FAIL  apps/dav/src/views/CalDavSettings.spec.js
  ● Test suite failed to run

    TypeError: Cannot destructure property 'config' of 'undefined' as it is undefined.

      1 | import { render } from '@testing-library/vue'
      2 | import userEvent from '@testing-library/user-event'
    > 3 | import CalDavSettings from './CalDavSettings'
        | ^
      4 | // eslint-disable-next-line no-unused-vars
      5 | import { generateUrl } from '@nextcloud/router'
      6 |

      at Object.getCacheKey (node_modules/vue-jest/lib/index.js:10:7)
      at ScriptTransformer._getCacheKey (node_modules/@jest/transform/build/ScriptTransformer.js:280:41)
      at ScriptTransformer._getFileCachePath (node_modules/@jest/transform/build/ScriptTransformer.js:351:27)
      at ScriptTransformer.transformSource (node_modules/@jest/transform/build/ScriptTransformer.js:588:32)
      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:758:40)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:815:19)
      at Object.<anonymous> (apps/dav/src/views/CalDavSettings.spec.js:3:1)
```

Also, it avoids using `jsdom-global`, which allows removing the `setupFilesAfterEnv` entry and makes the `jestBabelTransformer.js` slighly simpler by using the export from `babel-jest` directly instead of aliasing default.

**Review by artonge — APPROVED** [review]

**Pull request merged**